src/Interpreters/TemporaryReplaceTableName.cpp

```diff
@@ -0,0 +1,26 @@
+#include "TemporaryReplaceTableName.h"
+
+#include <Common/re2.h>
+
+#include <base/hex.h>
+#include <fmt/core.h>
+
+
+namespace DB
+{
+    String TemporaryReplaceTableName::toString() const
+    {
+        return fmt::format("_tmp_replace_{}_{}", name_hash, random_suffix);
+    }
+
+    std::optional<TemporaryReplaceTableName> TemporaryReplaceTableName::fromString(const String & str)
+    {
+        static const re2::RE2 pattern("_tmp_replace_(.*)_(.*)");
```

Comment:
Nice work adding the unit tests :tada: 
Two of them are failing, which is good because they were useful for finding issues on the current implementation.

The problem with this regex is that it can match an empty string (because of the `*`) if the suffix is missing.
We'd rather use `.+`, or even better, something that doesn't match everything, but only things we may expect: 

```suggestion
        static const re2::RE2 pattern(R"(^_tmp_replace_(\w+)_(\w+)$)");
```

With this, all unit tests should pass